features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/OtelTracerProvider.kt

```diff
@@ -229,11 +251,59 @@ class OtelTracerProvider(
         // endregion
     }
 
+    // region Internal
+
+    private fun resolveInstrumentationScopeName(instrumentationScopeName: String): String {
+        return if (instrumentationScopeName.trim { it <= ' ' }.isEmpty()) {
+            DEFAULT_TRACER_NAME
+        } else {
+            instrumentationScopeName
+        }
+    }
+
+    private fun resolveSpanBuilderDecorator(): (SpanBuilder) -> SpanBuilder {
+        return if (bundleWithRumEnabled) {
+            resolveSpanBuilderDecoratorFromContext()
+        } else {
+            NO_OP_SPAN_BUILDER_DECORATOR
+        }
+    }
+
+    private fun resolveSpanBuilderDecoratorFromContext(): (SpanBuilder) -> SpanBuilder = { spanBuilder ->
+        val rumContext = sdkCore.getFeatureContext(Feature.RUM_FEATURE_NAME)
+        val applicationId = rumContext[RUM_APPLICATION_ID_KEY] as? String
+        val sessionId = rumContext[RUM_SESSION_ID_KEY] as? String
+        val viewId = rumContext[RUM_VIEW_ID_KEY] as? String
+        val actionId = rumContext[RUM_ACTION_ID_KEY] as? String
+        if (applicationId != null && sessionId != null && viewId != null) {
+            spanBuilder.setAttribute(LogAttributes.RUM_APPLICATION_ID, applicationId)
+            spanBuilder.setAttribute(LogAttributes.RUM_SESSION_ID, sessionId)
+            spanBuilder.setAttribute(LogAttributes.RUM_VIEW_ID, viewId)
+            if (actionId != null) {
+                spanBuilder.setAttribute(LogAttributes.RUM_ACTION_ID, actionId)
+            }
+        } else {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                listOf(InternalLogger.Target.USER, InternalLogger.Target.TELEMETRY),
```

Comment:
is it useful in the telemetry?

Reply:
Good question. I was thinking that maybe it is useful for us to see how many times we are missing the link with RUM because RUM context is corrupted or session is not there. But I am opened to discussions here.

Reply:
My concern is just to avoid spamming telemetry / avoid having many telemetry events to process in the SDK itself, given that it is tracing, so we may have a lot of spans.

Reply:
I'll remove it in that case and we will see if we need it in case someone reports missing links.